gulpfile.babel.js

```diff
@@ -1,334 +1,45 @@
-			log( 'dops-components CSS finished.' );
-			doRTL( 'dops', done );
-		} );
-}
-
-function doRTL( files, done ) {
-	gulp.src( 'main' === files ? './_inc/build/style.min.css' : './_inc/build/*dops-style.css' )
-		.pipe( rtlcss() )
-		.pipe( rename( { suffix: '.rtl' } ) )
-		.pipe( sourcemaps.init() )
-		.pipe( sourcemaps.write( './' ) )
-		.pipe( gulp.dest( './_inc/build' ) )
-		.on( 'end', function() {
-			log( 'main' === files ? 'Dashboard RTL CSS finished.' : 'DOPS Components RTL CSS finished.' );
-			if ( done && 'function' === typeof done ) {
-				done();
-			}
-		} );
-}
-
-gulp.task( 'sass:build', [ 'react:build' ], doSass );
-
-gulp.task( 'sass:watch', function() {
-	doSass();
-	gulp.watch( [ './**/*.scss', ...alwaysIgnoredPaths ], doSass );
-} );
-
-gulp.task( 'react:build', function( done ) {
-	const config = getWebpackConfig();
-
-	if ( 'production' !== process.env.NODE_ENV ) {
-		config.plugins.push(
-			new webpack.LoaderOptionsPlugin( {
-				debug: true
-			} )
-		);
-	}
-
-	webpack( config ).run( onBuild( done ) );
-} );
-
-gulp.task( 'react:watch', function() {
-	const config = getWebpackConfig();
+module.exports = {
+	alwaysIgnoredPaths: alwaysIgnoredPaths
+};
 
-	webpack( config ).watch( 100, onBuild() );
-} );
-
-function doStatic( done ) {
-	let path;
-
-	const jsdom = require( 'jsdom' );
-
-	log( 'Building static HTML from built JS…' );
-
-	jsdom.env( '', function( err, window ) {
-		global.window = window;
-		global.document = window.document;
-		global.navigator = window.navigator;
-
-		window.Initial_State = {
-			dismissedNotices: [],
-			connectionStatus: {
-				devMode: {
-					isActive: false
-				}
-			},
-			userData: {
-				currentUser: {
-					permissions: {}
-				}
-			}
-		};
-
-		try {
-			path = __dirname + '/_inc/build/static.js';
-
-			delete require.cache[ path ]; // Making sure NodeJS requires this file every time this is called
-			require( path );
-
-			gulp.src( [ '_inc/build/static*' ] )
-				.pipe( tap( function( file ) {
-					fs.unlinkSync( file.path );
-				} ) )
-				.on( 'end', function() {
-					fs.writeFileSync( __dirname + '/_inc/build/static.html', window.staticHtml );
-					fs.writeFileSync( __dirname + '/_inc/build/static-noscript-notice.html', window.noscriptNotice );
-					fs.writeFileSync( __dirname + '/_inc/build/static-version-notice.html', window.versionNotice );
-					fs.writeFileSync( __dirname + '/_inc/build/static-ie-notice.html', window.ieNotice );
-
-					if ( done ) {
-						done();
-					}
-				} );
-		} catch ( error ) {
-			log( colors.yellow(
-				'Warning: gulp was unable to update static HTML files.\n\n' +
-				'If this is happening during watch, this warning is OK to dismiss: sometimes webpack fires watch handlers when source code is not yet built.'
-			) );
-		}
-	} );
-}
+import {} from './tools/builder/frontend-css';
+import {} from './tools/builder/admin-css';
+import {} from './tools/builder/react';
+import {} from './tools/builder/sass';
```

Comment:
:heart: the modularization!

Can we make the imports a bit less magical tho? :smile: What about

```difff
diff --git a/gulpfile.babel.js b/gulpfile.babel.js
index c87e3e6e3..31d79b54c 100644
--- a/gulpfile.babel.js
+++ b/gulpfile.babel.js
@@ -33,7 +33,7 @@ module.exports = {
 	alwaysIgnoredPaths: alwaysIgnoredPaths
 };
 
-import {} from './tools/builder/frontend-css';
+import frontendcss from './tools/builder/frontend-css';
 import {} from './tools/builder/admin-css';
 import {} from './tools/builder/react';
 import {} from './tools/builder/sass';
@@ -290,7 +290,7 @@ gulp.task( 'gutenpack:jetpack-blocks', function() {
 
 gulp.task(
 	'old-styles',
-	gulp.parallel( 'frontendcss', 'admincss', 'admincss:rtl', 'sass:old' )
+	gulp.parallel( frontendcss, 'admincss', 'admincss:rtl', 'sass:old' )
 );
 gulp.task( 'jshint', gulp.parallel( 'js:hint' ) );
 gulp.task( 'php', gulp.parallel( 'php:lint', 'php:unit' ) );
diff --git a/tools/builder/frontend-css.js b/tools/builder/frontend-css.js
index 55528191a..c39c0104f 100644
--- a/tools/builder/frontend-css.js
+++ b/tools/builder/frontend-css.js
@@ -75,7 +75,7 @@ const pathModifier = function( file, contents ) {
 };
 
 // Frontend CSS.  Auto-prefix and minimize.
-gulp.task( 'frontendcss', function() {
+export default function frontendcss() {
 	return gulp.src( concat_list )
 		.pipe( modify( { fileModifier: pathModifier } ) )
 		.pipe( autoprefixer(
@@ -101,9 +101,9 @@ gulp.task( 'frontendcss', function() {
 		.on( 'end', function() {
 			util.log( 'Front end modules CSS finished.' );
 		} );
-} );
+}
 
-gulp.task( 'frontendcss:separate', gulp.series( 'frontendcss', function() {
+gulp.task( 'frontendcss:separate', gulp.series( frontendcss, function() {
 	return gulp.src( separate_list )
 		.pipe( modify( { fileModifier: pathModifier } ) )
 		.pipe( autoprefixer(
```

See e.g. https://github.com/gulpjs/gulp/blob/master/docs/recipes/exports-as-tasks.md

Reply:
(Probably fine to do in a separate PR)